Source/monster.h

```diff
@@ -148,7 +148,7 @@ struct MonsterStruct { // note: missing field _mAFNum
 	int _menemy;
 	/** Usually correspond's to the enemy's future position */
 	Point enemyPosition;
-	byte *_mAnimData;
+	CelSprite *_mAnimData;

```

Comment:
I see this is declared as a pointer to avoid creating a copy of a CelSprite whenever the animation changes. It looks like making this a `std::reference_wrapper<std::optional<CelSprite>>` type would save having to constantly get and then take the address of the optional, while also allowing the removal of the two uses of `std::optional::has_value` in monster.cpp

Reply:
The design was introduced for items (see pr #1796). I follow that design.
This pr only unify monster implementation with items and players (That is a prerequisite for ADL).

Personally I would stick with a `shared_ptr` instead of a optional. But I don't have a strong opinion in that case.

Anyway I would rather not change this as part of this pr as this affects also items and players and that means it's outside of the scope of this. But that doesn't mean it can't be changed later. 🙂

Reply:
`shared_ptr` is a code smell in C++ most of the time -- every object should have an owner, it is rare that there really isn't a clear one.